### DISCUSSION TITLE
GraphQL::Schema::Argument::InvalidDefaultValueError

### DISCUSSION THREAD

**Discussion by therealrodk** [patterns-ai-core/langchainrb#253] [Q&A] [answered]
# The Code
I am trying to use this gem for the first time, and my code looks like this:
```ruby
class QueryService
  require "langchain"

  def initialize
    @client = Langchain::Vectorsearch::Weaviate.new(
      url: Rails.application.credentials.weaviate[:url],
      api_key: Rails.application.credentials.weaviate[:api_key],
      index_name: "langchain_test",
      llm: Langchain::LLM::OpenAI.new(api_key: Rails.application.credentials.openai[:api_key])
    )
    @client.create_default_schema
  end

  def add_texts(texts: [])
    @client.add_texts(
      texts: texts
    )
  end

  def ask(question)
    @client.ask(
      question: question
    )
  end

  def search(text)
    @client.similarity_search(
      query: text,
      k: 1
    )
  end
end
```

# The Issue
Running `qs = QueryService.new` or `qs.add_texts(texts: ['Bob likes cheese.', 'Marie likes soup.'])` works just fine. But when I try to query something like:

```ruby
qs.ask('What does Bob like?')
```
or
```ruby
qs.search('cheese')
```
I get the following error:

```bash
/Users/my_username/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/graphql-2.0.24/lib/graphql/schema/argument.rb:357:in `validate_default_value': `GetObjectsLangchain_testHybridInpObj.fusionType` has an invalid default value: `0` isn't accepted by `FusionEnum`; update the default value or the argument type. (GraphQL::Schema::Argument::InvalidDefaultValueError)
```

# The Question
I don't know anything about GraphQL and, as I said before, this gem is new to me. Am I doing something wrong? Or should I open an Issue?

**Comment by therealrodk** [comment]
I am using the latest version of this gem from RubyGems.org ([0.6.5](https://rubygems.org/gems/langchainrb)).

**Comment by andreibondarev** [comment]
@therealrodk When specifying the `index_name:`, could you please try capitalizing it?
```ruby
...
index_name: "Langchain_test"
...
```

Take a look at this note right here: https://github.com/andreibondarev/langchainrb/blob/main/lib/langchain/vectorsearch/weaviate.rb#L28-L30

**Comment by andreibondarev** [comment]
I think @obie was seeing a similar issue:
```
`GetObjectsLangchain_testHybridInpObj.fusionType` has an invalid default value: `0` isn't accepted by `FusionEnum`; update the default value or the argument type. (GraphQL::Schema::Argument::InvalidDefaultValueError)
```

**Comment by andreibondarev** [comment, marked as answer]
@therealrodk This issue was happening on the Weaviate side. They had fixed it and released a new version: 1.20.2. 

Please update your Weaviate version and try again!
